addons/sale_coupon/models/coupon_program.py

```diff
@@ -12,7 +12,7 @@ class CouponProgram(models.Model):
     # The api.depends is handled in `def modified` of `sale_coupon/models/sale_order.py`
     def _compute_order_count(self):
         for program in self:
-            program.order_count = self.env['sale.order.line'].search_count([('product_id', '=', program.discount_line_product_id.id)])
+            program.order_count = self.env['sale.order.line'].sudo().search_count([('product_id', '=', program.discount_line_product_id.id)])
```

Comment:
Or set the count to 0 if the user doesn't have read access to SOL ?

Reply:
@Feyensv Having a count to 0 will lead to error when you are using the "total_order_count" field to know if you have used the program more than the "maximum_use_number".